apps/src/lib/levelbuilder/reference-guide-editor/ReferenceGuideEditAll.jsx

```diff
@@ -0,0 +1,187 @@
+import React, {useMemo, useCallback, useState} from 'react';
+import PropTypes from 'prop-types';
+import $ from 'jquery';
+import {TextLink} from '@dsco_/link';
+import FontAwesome from '@cdo/apps/templates/FontAwesome';
+import {flatten} from 'lodash';
+
+// editAll url without the /edit
+const BASE_URL = window.location.href
+  .split('/')
+  .slice(0, -1)
+  .join('/');
+
+const MiniIconButton = ({icon, alt, func, href}) => (
+  <TextLink
+    onClick={func}
+    href={href}
+    icon={<FontAwesome icon={icon} title={alt} />}
+  />
+);
+MiniIconButton.propTypes = {
+  icon: FontAwesome.propTypes.icon,
+  alt: PropTypes.string,
+  func: PropTypes.func,
+  href: PropTypes.string
+};
+
+const getGuideChildren = (key, guides) =>
+  guides
+    .filter(guide => guide.parent_reference_guide_key === key)
+    .sort((a, b) => a.position - b.position);
+
+// take a list of reference guides and build a tree data structure and then flatten it
+const organizeReferenceGuides = (referenceGuides, parent = null, level = 0) => {
+  const organizedGuides = getGuideChildren(parent, referenceGuides).map(
+    guide => [
+      {...guide, level}, // add the depth of this guide so we can render the indentation
+      ...organizeReferenceGuides(referenceGuides, guide.key, level + 1) // put the children right after the parent
+    ]
+  );
+  return flatten(organizedGuides);
+};
+
+export default function ReferenceGuideEditAll(props) {
+  const [referenceGuides, setReferenceGuides] = useState(props.referenceGuides);
+  // useMemo here so that we only do the organizing once
+  const organizedGuides = useMemo(
+    () => organizeReferenceGuides(referenceGuides),
+    [referenceGuides]
+  );
+
+  const deleteGuide = useCallback(
+    guideKey => {
+      setReferenceGuides([
+        ...referenceGuides.filter(guide => guide.key !== guideKey)
+      ]);
+      fetch(`${BASE_URL}/${guideKey}`, {
```

Comment:
What happens if you delete a guide that is the parent of other guides?

Reply:
goood question, right now the children would become orphans (still in the db, but not visible/linked to from anywhere). I'll add a thing to clean those up. I can think of a few options on how to handle that, so i'll check with tess

Reply:
ok, I think the plan is to delete the child ref guides, but since I already have all of the information on the page, I'll add a confirmation warning dialog and list out all the reference guides that would be affected by doing that.